File: src/test/java/compiler/block/primitives/FloatBlock.java
package compiler.block.primitives;

import compiler.block.Block;

public class FloatBlock extends Block {

    private String type = "float";
    private String value;
    private String name;

    public FloatBlock(Block superBlock, String name, String value) {
        super(superBlock, false, true);
        this.name = name;
        this.value = value;
    }

    @Override
    public void init() {

    }

    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return "";
    }

    @Override
    public String getClosingCode() {
        return "";
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/scala/compiler/block/primitives/FloatBlock.scala
package compiler.block.primitives

import compiler.block.Block

class FloatBlock(superBlockInit: Block, declaration : Boolean, name: String, value: String) extends Block(superBlockInit, false, true) {

  override def init() {}

  override def getName: String = {
    name
  }

  override def getValue: String = {
    value
  }

  override def getType: String = {
    "float"
  }

  override def getOpeningCode: String = {
    asm.visitLdcInsn("new Float(" + value + ")") +
      asm.visitVarInsn("FSTORE", id);
  }

  override def getClosingCode: String = {
    ""
  }

  override def toString: String = "float: " + name + " = " + value

}